File: Springboot/bintobloom-backend/src/main/java/com/bintobloom/entity/TrackingLog.java
package com.bintobloom.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.time.LocalDateTime;
import java.math.BigDecimal;

@Entity
@Table(name = "tracking_logs")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class TrackingLog {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "track_id")
    private Long trackId;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "pickup_id", nullable = false)
    private PickupRequest pickupRequest;
    
    @Column(nullable = false, precision = 10, scale = 8)
    private BigDecimal latitude;
    
    @Column(nullable = false, precision = 11, scale = 8)
    private BigDecimal longitude;
    
    @Column(nullable = false)
    private LocalDateTime timestamp;
}

File: .NET/BinToBloom Backend/Models/Entities/TrackingLog.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BinToBloom_Backend.Models.Entities
{
    [Table("TrackingLogs")]
    public class TrackingLog
    {
        [Key]
        [Column("track_id")]
        public int TrackId { get; set; }

        [Required]
        [Column("pickup_id")]
        public int PickupId { get; set; }

        [Required]
        [Column("latitude")]
        [Range(-90, 90, ErrorMessage = "Latitude must be between -90 and 90")]
        public decimal Latitude { get; set; }

        [Required]
        [Column("longitude")]
        [Range(-180, 180, ErrorMessage = "Longitude must be between -180 and 180")]
        public decimal Longitude { get; set; }

        [Column("timestamp")]
        public DateTime Timestamp { get; set; } = DateTime.Now;

        [ForeignKey("PickupId")]
        public PickupRequest PickupRequest { get; set; } = null!;
    }
}